File: leetcode/0828-count-unique-characters-of-all-substrings-of-a-given-string/uniqueLetterString_test.go
package leetcode

import (
	"fmt"
	"path/filepath"
	"testing"

	"github.com/edram/leetcode/util"
)

type UniqueLetterStringCase struct {
	Input string

	Output int
}

func TestUniqueLetterString(t *testing.T) {
	var casePath, _ = filepath.Abs("./cases.json")
	payload, err := util.LoadCases[[]UniqueLetterStringCase](casePath)
	if err != nil {
		panic(err)
	}

	for i, testCase := range *payload {
		fmt.Printf("正在执行第 %d 个测试用例\n", i+1)
		fmt.Printf("nums: %v ==> %v \n", testCase.Input, testCase.Output)

		result := uniqueLetterString(testCase.Input)
		if result != testCase.Output {
			t.Errorf("期望 %v, 结果 %v", testCase.Output, result)
		}
	}
}

File: leetcode/0828-count-unique-characters-of-all-substrings-of-a-given-string/uniqueLetterString.test.ts
import cases from "./cases.json";
import uniqueLetterString from "./uniqueLetterString";

test("count unique characters of all substrings of a given string", () => {
  cases.forEach((testCase, index) => {
    console.log(`正在执行第 ${index + 1} 个测试用例\n`);
    expect(uniqueLetterString(testCase.input)).toEqual(testCase.output);
  });
});
